### PR TITLE
9943 - Add new spacing units on Tailwind

### PR DESCRIPTION
**Pull request by fessehaye**
# Description

Probably easier to follow commit by commit!

Complete overhaul of tailwind spacing units we are using. This includes overriding and including new tokens.

`en/style-guide` spacing section is updated accordingly. If you have tickets that are using or updating width/margin/padding etc, this PR will probably affect your PRs after being merged.

Also was able to replace some scss lines with tailwind classes to make future refactors of the sass files a bit easier to remember which classes to use.

```
By default, these values are inherited by the padding, margin, width, height, maxHeight, flex-basis, gap, inset, space, translate, scrollMargin, scrollPadding, and textIndent core plugins.
``` 

Related PRs/issues: #9943 

## Current implementation

| Tailwind Unit  | Bootstrap Unit | Size (rem) |
| --------------- | ---------------- | ----------- |
| 1                     | 1                      | 0.25rem     |
| 2                     | 2                      | 0.50rem     |
| 3                     |                         | 0.75rem     |
| 4                     | 3                      | 1rem         |
| 5                     | 4                     | 1.5rem       |
| 6                     |                        | 2rem         |
| 7                     | 5                     | 3rem         |

## Current Design Tokens

https://www.figma.com/file/JT9Dvdyc96AqQ3HJlnA9Qq/WIP-Mozilla-Foundation-Design-Style-Library

## Converting Guide

| Tailwind Unit  | Old Size           | New Size   |
| --------------- | ---------------- | ----------- |
| 0                     | 0                      | 0               |
| px                   | 1px                  | 1px           |
| rem                |                         | 1rem        |
| 1                     | 0.25rem           | 2px         |
| 2                     | 0.5rem             | 4px         |
| 3                     | 0.75rem           | 6px         |
| 4                     | 1rem                | 8px         |
| 5                     | 1.5rem             | 10px       |
| 6                     | 2rem                | 12px       |
| 7                     | 3rem                | 14px       |
| 8                     |                         | 16px       |
| 9                     |                         | 18px       |
| 10                   |                         | 20px       |
| 12                   |                         | 24px       |
| 14                   |                         | 28px       |
| 16                   |                         | 32px       |
| 18                   |                         | 36px       |
| 20                   |                         | 40px       |
| 22                   |                         | 44px       |
| 24                   |                         | 48px       |
| 28                   |                         | 56px       |
| 32                   |                         | 64px       |
| 40                   |                         | 80px       |
| 48                   |                         | 96px       |
| 56                   |                         | 112px     |
| 64                   |                         | 128px     |
| 72                   |                         | 144px     |
| 80                   |                         | 160px     |
| 88                   |                         | 176px     |
| 96                   |                         | 192px     |
| 104                 |                         | 208px     |
| 112                 |                         | 224px     |
| 120                 |                         | 240px     |
| 128                 |                         | 256px     |
| 144                 |                         | 288px     |
| 160                 |                         | 320px     |
| 192                 |                         | 384px     |

### CODE REVIEW

File: network-api/networkapi/wagtailpages/templates/wagtailpages/fragments/styleguide_spacing_table.html [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+                <tr class="tw-bg-black tw-bg-opacity-5">
+                    <td class="tw-p-6 tw-border tw-border-table tw-font-bold">144</td>
+                    <td class="tw-p-6 tw-border tw-border-table">288px (18rem)</td>
+                    <td class="tw-p-6 tw-border tw-border-table"><div class="tw-bg-red-40 tw-w-144 tw-h-144"></div></td>
+                </tr>
+                <tr class="">
+                    <td class="tw-p-6 tw-border tw-border-table tw-font-bold">160</td>
+                    <td class="tw-p-6 tw-border tw-border-table">320px (20rem)</td>
+                    <td class="tw-p-6 tw-border tw-border-table"><div class="tw-bg-red-40 tw-w-160 tw-h-160"></div></td>
+                </tr>
```
Comment (mmmavis) [inline network-api/networkapi/wagtailpages/templates/wagtailpages/fragments/styleguide_spacing_table.html:191]: Can we add a row for `192` as well? Thanks!

File: network-api/networkapi/wagtailpages/templates/wagtailpages/fragments/styleguide_spacing_table.html [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+                    <td class="tw-p-6 tw-border tw-border-table"><div class="tw-bg-red-40 tw-w-40 tw-h-40"></div></td>
+                </tr>
+                <tr class="">
+                    <td class="tw-p-6 tw-border tw-border-table tw-font-bold">48</td>
+                    <td class="tw-p-6 tw-border tw-border-table">64px (6rem)</td>
+                    <td class="tw-p-6 tw-border tw-border-table"><div class="tw-bg-red-40 tw-w-48 tw-h-48"></div></td>
+                </tr>
+                <tr class="tw-bg-black tw-bg-opacity-5">
+                    <td class="tw-p-6 tw-border tw-border-table tw-font-bold">56</td>
+                    <td class="tw-p-6 tw-border tw-border-table">80px (7rem)</td>
```
Comment (mmmavis) [inline network-api/networkapi/wagtailpages/templates/wagtailpages/fragments/styleguide_spacing_table.html:134]:
```suggestion
                    <td class="tw-p-6 tw-border tw-border-table">112px (7rem)</td>
```

File: network-api/networkapi/wagtailpages/templates/wagtailpages/fragments/styleguide_spacing_table.html [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+                    <td class="tw-p-6 tw-border tw-border-table"><div class="tw-bg-red-40 tw-w-32 tw-h-32"></div></td>
+                </tr>
+                <tr class="tw-bg-black tw-bg-opacity-5">
+                    <td class="tw-p-6 tw-border tw-border-table tw-font-bold">40</td>
+                    <td class="tw-p-6 tw-border tw-border-table">80px (5rem)</td>
+                    <td class="tw-p-6 tw-border tw-border-table"><div class="tw-bg-red-40 tw-w-40 tw-h-40"></div></td>
+                </tr>
+                <tr class="">
+                    <td class="tw-p-6 tw-border tw-border-table tw-font-bold">48</td>
+                    <td class="tw-p-6 tw-border tw-border-table">64px (6rem)</td>
```
Comment (mmmavis) [inline network-api/networkapi/wagtailpages/templates/wagtailpages/fragments/styleguide_spacing_table.html:129]:
```suggestion
                    <td class="tw-p-6 tw-border tw-border-table">96px (6rem)</td>
```

File: network-api/networkapi/wagtailpages/templates/wagtailpages/pages/youtube-regrets-2022/fragments/timeline.html [hunk 4]
```diff
@@ -249,7 +249,7 @@ <h2>{% trans 'Three months later' %}</h2>
 
 {# Create curved effect on section using border radius #}
 <div class="tw-rounded-b-[30px] tw-mb-[-30px]">
-    <section class="container tw-max-w-4xl tw-text-center tw-my-7 large:tw-my-[80px] large:tw-my-[120px]" data-scroll-section>
+    <section class="container tw-max-w-4xl tw-text-center tw-my-24 large:tw-my-[120px]" data-scroll-section>
```
Comment (mmmavis) [inline network-api/networkapi/wagtailpages/templates/wagtailpages/pages/youtube-regrets-2022/fragments/timeline.html:252]: Reminder for @mmmavis : check if we should be keeping `large:tw-my-[80px]` or `large:tw-my-[120px]`

File: network-api/networkapi/wagtailpages/templates/wagtailpages/campaign_page.html [hunk 5]
```diff
@@ -56,18 +56,18 @@ <h1 class="tw-h1-heading {% if uses_menu %}tw-hidden medium:tw-block{% endif %}"
...
                                     tw-w-full
                                 {% endif %}"
                             >
                                 <div class="narrow-sticky-button-container hidden">
                                     <a href="#cta-anchor" class="tw-btn-primary tw-block">{% trans "TAKE ACTION" context "Sticky button in mobile view on campaign pages" %}</a>
                                 </div>
-                                <div class="tw-sticky tw-top-[5rem] tw-overflow-hidden tw-z-[1018]">
-                                    <div id="cta-anchor" class="tw-mt-[-80px] tw-pt-[80px]">
+                                <div class="tw-sticky tw-top-40 tw-overflow-hidden tw-z-[1018]">
+                                    <div id="cta-anchor" class="tw-mt--40 tw-pt-40">
```
Comment (mmmavis) [inline network-api/networkapi/wagtailpages/templates/wagtailpages/campaign_page.html:70]:
```suggestion
                                    <div id="cta-anchor" class="-tw-mt-40 tw-pt-40">
```

**Review by mmmavis — CHANGES_REQUESTED** [review]
I'm only 2/3 way down the commit list (up till spacing 8 changes). Gonna submit the issues I found so far and review the last 1/3 after lunch.

**Review by mmmavis — CHANGES_REQUESTED** [review]
Just happened to spot this - PNI search icon on search bar is smaller than before

image

File: network-api/networkapi/wagtailpages/templates/wagtailpages/pages/youtube-regrets-2022/fragments/timeline.html (hunk 4 above)
Comment (mmmavis) [reply network-api/networkapi/wagtailpages/templates/wagtailpages/pages/youtube-regrets-2022/fragments/timeline.html:252]: I did a quick test by adding `large:tw-my-[80px] large:tw-my-[120px]` to a `<div>`. It ends up having top/bottom margin of 80px. In this case I think we should be keeping `large:tw-my-[80px]` instead.

```suggestion
    <section class="container tw-max-w-4xl tw-text-center tw-my-24 large:tw-my-40" data-scroll-section>
```

**Review by mmmavis — DISMISSED** [review]

**Comment by fessehaye** [comment]
> Just happened to spot this - PNI search icon on search bar is smaller than before
> 
> image

Fixed

File: network-api/networkapi/wagtailpages/templates/wagtailpages/fragments/styleguide_spacing_table.html (hunk 1 above)
Comment (fessehaye) [reply network-api/networkapi/wagtailpages/templates/wagtailpages/fragments/styleguide_spacing_table.html:191]: http://localhost:8000/en/style-guide/#spacing is updated now

**Comment by fessehaye** [comment]
@mmmavis should be good to review again

**Review by danielfmiranda — APPROVED** [review]
Hey @fessehaye thanks for taking this on! It definitely looks like it was a lot of work but will be helpful in the future.

After taking a look at the commits that werent yet reviewed and also taking a look at the site on my local machine. everything is looking good to me.

Approved!

**Pull request merged**